nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -341,6 +341,33 @@ def run(config):
     address = os.environ.get("REDIS_ADDRESS", config.get("redis_address"))
     ray.init(address=address, local_mode=config.get("local_mode", False))
 
+    # Register serializer and deserializer - needed when logging arrays and tensors.
```

Comment:
The docs say:
The serializer and deserializer are used when transferring objects of `cls` across processes and nodes. This can be significantly faster than the Ray default fallbacks. Wraps `register_custom_serializer` underneath.

It's thus not used exclusively by the loggers, but it is invoked when the logs have array-like objects. In practice, I've only seen them used on the logs though.

Reply:
Did you test with distributed training using multiple GPU nodes?

Reply:
I did. That was the main case that inspired the addition. What's your line of thinking?

Reply:
I am concerned about `obj.cpu().numpy()` call. We just need to make sure we understand the performance cost of adding this extra serialization. If you tested with multiple nodes and the serialization was only used by the loggers and you did not notice any performance loss then it should be fine

Reply:
That's a fair concern. I can attempt a test today to double check. I'm 90% sure, but you're right in the care needed to keep performance up.